Sources/NIOHTTP2/HTTP2Settings.swift

```diff
@@ -12,101 +12,232 @@
+    /// this value. It is recommended that this value be no smaller than 100, so as
+    /// to not unnecessarily limit parallelism.
+    ///
+    /// A value of 0 is legal, and will prevent the creation of new streams.
+    case maxConcurrentStreams(Int32)
 
-    /// Corresponds to SETTINGS_MAX_CONCURRENT_STREAMS
-    public static let maxConcurrentStreams = HTTP2SettingsParameter(3)
+    /// SETTINGS_INITIAL_WINDOW_SIZE (0x4)
+    ///
+    /// Indicates the sender's initial window size (in octets) for stream-level
+    /// flow control. The initial value is 2^16-1 (65,535) octets.
+    ///
+    /// Values above the maximum flow-control window size of 2^31-1 are illegal,
+    /// and will result in a connection error.
+    case initialWindowSize(Int32)
 
-    /// Corresponds to SETTINGS_INITIAL_WINDOW_SIZE
-    public static let initialWindowSize = HTTP2SettingsParameter(4)
+    /// SETTINGS_MAX_FRAME_SIZE (0x5)
+    ///
+    /// Indicates the size of the largest frame payload that the sender is willing
+    /// to receive, in octets.
+    ///
+    /// The initial value is 2^14 (16,384) octets.  The value advertised by an
+    /// endpoint MUST be between this initial value and the maximum allowed frame
+    /// size (2^24-1 or 16,777,215 octets), inclusive.
+    case maxFrameSize(Int32)
 
-    /// Corresponds to SETTINGS_MAX_FRAME_SIZE
-    public static let maxFrameSize = HTTP2SettingsParameter(5)
+    /// SETTINGS_MAX_HEADER_LIST_SIZE (0x6)
+    ///
+    /// This advisory setting informs a peer of the maximum size of header list
+    /// that the sender is prepared to accept, in octets.  The value is based on the
+    /// uncompressed size of header fields, including the length of the name and
+    /// value in octets plus an overhead of 32 octets for each header field.
+    case maxHeaderListSize(Int32)
 
-    /// Corresponds to SETTINGS_MAX_HEADER_LIST_SIZE
-    public static let maxHeaderListSize = HTTP2SettingsParameter(6)
-}
+    /// SETTINGS_ACCEPT_CACHE_DIGEST (0x7)
+    ///
+    /// A server can notify its support for CACHE_DIGEST frame by sending this
+    /// parameter with a value of `true`. If the server is tempted to make
+    /// optimizations based on CACHE_DIGEST frames, it SHOULD send this parameter
+    /// immediately after the connection is established.
+    case acceptCacheDigest(Bool)
 
-extension HTTP2SettingsParameter: Equatable {
-    public static func ==(lhs: HTTP2SettingsParameter, rhs: HTTP2SettingsParameter) -> Bool {
-        return lhs.networkRepresentation == rhs.networkRepresentation
+    /// SETTINGS_ENABLE_CONNECT_PROTOCOL (0x8)
+    ///
+    /// Upon receipt of this parameter with a value of `true`, a client MAY use
+    /// the Extended CONNECT method when creating new streams, for example to
+    /// bootstrap a WebSocket connection. Receipt of this parameter by a server
+    /// does not have any impact.
+    case enableConnectProtocol(Bool)
```

Comment:
Let's not add settings without an RFC number yet.

Reply:
This actual setting is the entire reason I'm working on HTTP/2 😉

I'm trying to get conforming reference implementations in place to nudge the RFC editor to actually look at the damned thing, since they've been sitting on it for a couple of months now with no action beyond a single question and answer right when it was first put into their queue. The same goes for much of the other items, too: most everything else on HTTP/2 the http-bis WG has completed & submitted to the RFC editors has basically sat dormant with no discussion while those of us who have real actual needs to use it now are holding our breath for it to be published. Hence muggins here trying to come up with a way for our iOS apps to tunnel a WebSocket through an HTTP/2 Adobe Traffic Server proxy to a back-end running a Netty or Akka HTTP/2 server: literally the one use-case that stops us enabling HTTP/2 across the organization. Sigh.

Ultimately I want my branch to be somewhere that I can put together experimental implementations of in-progress standards, to better provide feedback to the IETF & co. WebSocket bootstrapping, origin, and cache digests, however, are things I need to provide an implementation for asap, though, and actually form the primary focus of my work on the codecs and framers. Can you suggest a good way to handle this? Maintaining two parallel branches with this much churn on the implementation details is a real pain, but I guess that's my only option?

Reply:
Hm, actually, it has an RFC number assigned: RFC8441. It's just waiting on IANA registration of the "websocket" value for the `:protocol` header: https://www.rfc-editor.org/auth48/rfc8441

Reply:
This may be a good time to think about what, if anything, the extensibility model should be here. We'd really like to be able to support extensions without requiring that core NIO adopt them directly. In the short term I'm ok with putting this in, so long as we're confident that the various IANA-assigned code points aren't going to change. Given that this has an RFC number we're probably safe.

Reply:
Actually designing the extension point is a bit tricky, because we have a performance/flexibility trade-off that has to be made.